### PR TITLE
Vanilla Conquer Sdl 1.2

### PR DESCRIPTION
**Pull request by arczi84**
Initial SDL 1.2 support.
Only VanillaTD.

### CODE REVIEW

File: common/video_sdl1.h [hunk 1]
```diff
@@ -0,0 +1,111 @@
+//
+// SDL2 Backports
+//
+
```
Comment (giulianobelinassi) [inline common/video_sdl1.h:6]: Missing headerguard.
```
#ifndef VIDEO_SDL1_H
#define VIDEO_SDL1_H
```

File: common/video_sdl1.h [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	}
+	palette->colors = (SDL_Color *)SDL_malloc(ncolors * sizeof(*palette->colors));
+	if (!palette->colors) {
+		SDL_free(palette);
+		return NULL;
+	}
+	palette->ncolors = ncolors;
+	SDL_memset(palette->colors, 0xFF, ncolors * sizeof(*palette->colors));
+	return palette;
+}
```
Comment (giulianobelinassi) [inline common/video_sdl1.h:105]: `#endif // VIDEO_SDL1_H`

File: common/video_sdl1.h [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+{
+	if (!palette) {
+		SDL_InvalidParamError("palette");
+		return;
+	}
+	SDL_free(palette->colors);
+	//SDL_free(palette);
+}
+
+int
```
Comment (giulianobelinassi) [inline common/video_sdl1.h]: All these functions should be declared `static`. I assume they won't be visible in any compilation unit other than `video_sdl2.cpp`.

File: common/video_sdl1.h [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        SDL_SetError("SDL_SetPixelFormatPalette() passed a palette that doesn't match the format");
+		return -1;
+    }
+
+    if (format->palette == palette) {
+        return 0;
+    }
+
+    if (format->palette) {
+        //SDL_FreePalette(format->palette);
```
Comment (giulianobelinassi) [inline common/video_sdl1.h]: This if can be simply removed.

File: common/video_sdl1.h [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+inline void
+SDL_FreePalette(SDL_Palette *palette)
+{
+	if (!palette) {
+		SDL_InvalidParamError("palette");
+		return;
+	}
+	SDL_free(palette->colors);
+	//SDL_free(palette);
```
Comment (giulianobelinassi) [inline common/video_sdl1.h]: Why is this code commented out? If they serve no purpose, it can simply be deleted.

File: common/video_sdl1.h [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+// SDL2 Backports
+//
+
+#include <SDL.h>
+
+#define SDL_InvalidParamError(x)	   SDL_SetError("Parameter '%s' is invalid", (x))
+#define SDL_WINDOW_FULLSCREEN_DESKTOP  SDL_FULLSCREEN
+#define SDL_WINDOW_FULLSCREEN  		   SDL_FULLSCREEN
+
+inline void
```
Comment (giulianobelinassi) [inline common/video_sdl1.h]: Why are you using GNU-style C function declarations? please run
```
clang-format -i <file.h>
```
for it to be automatically formated according to this project coding style.

File: common/video_sdl1.h [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+    if (colors != (palette->colors + firstcolor)) {
+        SDL_memcpy(palette->colors + firstcolor, colors,
+                   ncolors * sizeof(*colors));
+    }
+
+    return status;
+}
+
+inline SDL_Palette *
```
Comment (giulianobelinassi) [inline common/video_sdl1.h]: Even the `inline` functions should be declared `static`, as it avoids generating symbols for it and duplicated code.

File: common/video_sdl2.cpp [hunk 8]
```diff
@@ -47,8 +47,14 @@
```
Comment (giulianobelinassi) [inline common/video_sdl2.cpp:47]: I guess this file should be renamed to `video_sdl.cpp` if they now contain code for both sdl1 and sdl2.

File: common/video_sdl2.cpp [hunk 9]
```diff
@@ -115,16 +122,21 @@ Uint32 SettingsPixelFormat()
...
 static void Update_HWCursor_Settings()
 {
     /*
     ** Update mouse scaling settings.
     */
+#ifdef SDL2_BUILD	
     int win_w, win_h;
     SDL_GetRendererOutputSize(renderer, &win_w, &win_h);
+#else
+	int win_w = 640;
```
Comment (giulianobelinassi) [inline common/video_sdl2.cpp]: Incorrect indentation. Also I think you should only guard the `SDL_GetRendererOutputSize` if sdl2 build, and default the variable to 640x400.

File: common/video_sdl2.cpp [hunk 10]
```diff
@@ -226,6 +240,10 @@ bool Set_Video_Mode(int w, int h, int bits_per_pixel)
         Settings.Video.WindowHeight = win_h;
     }
 
+#ifndef SDL2_BUILD
+    	window = SDL_SetVideoMode(w, h, 16/*bits_per_pixel*/, SDL_HWSURFACE);
```
Comment (giulianobelinassi) [inline common/video_sdl2.cpp]: I assume 16bpp isn't the most used video mode around? This should be read from a configuration file, and default to 24bpp if none was provided.

You could also default according to the operating system. I guess for Windows 95/.../XP, 8bpp is safe. For later versions, 32bpp is what most people use.

File: common/video_sdl2.cpp [hunk 11]
```diff
@@ -810,12 +856,18 @@ class VideoSurfaceSDL2 : public VideoSurface
         SDL_RenderClear(renderer);
         SDL_RenderCopy(renderer, texture, NULL, &render_dst);
         SDL_RenderPresent(renderer);
+#else	
+		SDL_BlitSurface(surface, NULL, window, NULL);
```
Comment (giulianobelinassi) [inline common/video_sdl2.cpp]: Incorrect indentation.

**Review by giulianobelinassi — CHANGES_REQUESTED** [review]

File: common/video_sdl2.cpp (hunk 10 above)
Comment (hifi) [reply common/video_sdl2.cpp]: I'd make the PixelFormat setting an ifdef between SDL1 and SDL2 adding a BitDepth for the former in settings.

**Comment by giulianobelinassi** [comment]
I had to apply the following patch for it to compile:

[0001-Add-cmake-script-for-finding-SDL1.zip](https://github.com/TheAssemblyArmada/Vanilla-Conquer/files/7414144/0001-Add-cmake-script-for-finding-SDL1.zip)

Also, this happens when I execute vanillatd:

Screenshot from 2021-10-25 20-53-21

File: common/video_sdl1.h (hunk 5 above)
Comment (arczi84) [reply common/video_sdl1.h]: They are present in sdl2 code but crash here.

**Pull request closed without merging**